base_export_async/models/delay_export.py

```diff
@@ -0,0 +1,128 @@
+
+        model_name, fields_name, ids, domain, import_compat, context = \
+            operator.itemgetter('model', 'fields', 'ids',
+                                'domain', 'import_compat', 'context')(params)
+        user = self.env['res.users'].browse([context.get('uid')])
+        if not user or not user.email:
+            raise UserError(_("The user doesn't have an email address."))
+
+        model = self.env[model_name].with_context(
+            import_compat=import_compat, **context)
+        records = model.browse(ids) or model.search(
+            domain, offset=0, limit=False, order=False)
+
+        if not model._is_an_ordinary_table():
+            fields_name = [field for field in fields_name
+                           if field['name'] != 'id']
+
+        field_names = [f['name'] for f in fields_name]
+        import_data = records.export_data(
+            field_names, raw_data).get('datas', [])
+
+        if import_compat:
+            columns_headers = field_names
+        else:
+            columns_headers = [val['label'].strip() for val in fields_name]
+
+        if export_format == 'csv':
+            csv = CSVExport()
+            return csv.from_data(columns_headers, import_data)
+        else:
+            xls = ExcelExport()
+            return xls.from_data(columns_headers, import_data)
+
+    @api.model
+    @job
+    def export(self, params):
+        content = self._get_file_content(params)
+
+        model_name, context, export_format = \
+            operator.itemgetter('model', 'context', 'format')(params)
+        user = self.env['res.users'].browse([context.get('uid')])
+
+        export_record = self.sudo().create({'user_id': user.id})
+
+        name = "{}.{}".format(model_name, export_format)
+        attachment = self.env['ir.attachment'].create({
+            'name': name,
+            'datas': base64.b64encode(content),
+            'datas_fname': name,
+            'type': 'binary',
+            'res_model': self._name,
+            'res_id': export_record.id,
+        })
+
+        url = "{}/web/content/ir.attachment/{}/datas/{}?download=true".format(
+            self.env['ir.config_parameter'].sudo().get_param('web.base.url'),
+            attachment.id,
+            attachment.name,
+        )
+
+        time_to_live = self.env['ir.config_parameter'].sudo(). \
+            get_param('attachment.ttl', 7)
+        date_today = fields.Date.today()
+        expiration_date = fields.Date.to_string(
+            date_today + relativedelta(days=+int(time_to_live)))
+
+        # TODO : move to email template
+        odoo_bot = self.sudo().env.ref("base.partner_root")
+        email_from = odoo_bot.email
+        self.env['mail.mail'].create({
+            'email_from': email_from,
+            'reply_to': email_from,
+            'email_to': user.email,
+            'subject': _("Export {} {}").format(
+                model_name, fields.Date.to_string(fields.Date.today())),
```

Comment:
@apineux last thing. Here could you use the model description (in the user language if it is translatable). It will be nicer for the user to see Invoice rather than account.invoice in the email subject.